client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -949,6 +951,12 @@ public final void fieldCapsAsync(FieldCapabilitiesRequest fieldCapabilitiesReque
             FieldCapabilitiesResponse::fromXContent, listener, emptySet());
     }
 
+    /**
+     * @deprecated If creating a new HLRC ReST API call, consider creating new actions instead of reusing server actions. The Validation
+     * layer has been added to the ReST client, and requests should extend
+     * {@link org.elasticsearch.protocol.xpack.common.ValidationException} instead of `ActionRequest`.
+     */
+    @Deprecated
```

Comment:
aren't we deprecating the way that all our current API are implemented?

Reply:
sadly, yes, if we want to use new server agnostic classes, we have to stop using the old ActionRequest classes.

Reply:
I agree with the direction, reusing classes is what got us far relatively quickly in the beginning, but as that becomes harder and harder, it makes little sense to continue doing so especially given that it gets us only to an intermediate result that still depends on server. I think as long as we don't deprecate methods that are already out, and we only deprecate what they use internally, this is ok. Do we also plan on rewriting the currently supported API with this approach sometimes soon?